internal/ipam/ipam.go

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package ipam
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/apparentlymart/go-cidr/cidr"
+)
+
+type Veths struct {
+	HostSide      net.IPNet
+	ContainerSide net.IPNet
+}
+
+// VethIPs returns the IPs for the host side and the PE side
+// for a given pool on the ith node.
+func VethIPs(pool string, index int) (Veths, error) {
```

Comment:
a bit unrelated (without the whole context), where would the index come from (a node will know it is the x node?)